js/index.js

```diff
@@ -2,16 +2,42 @@ var React = require("react");
 var ReactDOM = require("react-dom");
 
 var Navbar = require("./components/Navbar.js");
+var SubmitModal = require("./components/SubmitModal.js");
+var PostList = require("./components/PostList.js");
 
 class App extends React.Component {
 	constructor(props) {
 		super(props);
-		this.state = {};
+		this.handleNavbarRequest = this.handleNavbarRequest.bind(this);
+		this.handleCloseSubmitModal = this.handleCloseSubmitModal.bind(this);
+		this.state = {
+			showSubmitModal: false
+		};
```

Comment:
Still considering putting the database logic out of the React component. But certainly will avoid creating too many levels of "smart" components.